### PR TITLE
refactor rh_cloud tests

### PR DESCRIPTION
**Pull request by jameerpathan111**
This PR:
- removes `test_rh_cloud_tag_values` and `test_positive_tag_values_max_length` and covers test scenarios in `test_include_parameter_tags_setting`.
- removes `test_system_purpose_sla_field` and covers test scenario in `test_rhcloud_inventory_api_e2e`.
- removes `tests/foreman/cli/test_rhcloud_insights.py` and covers `test_positive_connection_option` in `configure_rhai_client` helper.
- removes `test_allow_auto_insights_sync_setting` as it's covered by `test_rhcloud_inventory_auto_upload_setting`.

### CODE REVIEW

**Comment by jameerpathan111** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_rhcloud_inventory.py      
```

**Review by adarshdubey-star — APPROVED** [review]
Clean refactor @jameerpathan111

File: robottelo/hosts.py [hunk 1]
```diff
@@ -1158,6 +1158,8 @@ def configure_rhai_client(
             # Register client
             if self.execute('insights-client --register').status != 0:
                 raise ContentHostError('Unable to register client to Insights through Satellite')
+            if self.execute('insights-client --test-connection').status != 0:
+                raise ContentHostError('insights-client --test-connection failed.')
```
Comment (adarshdubey-star) [inline robottelo/hosts.py:1162]: Non blocking:
```suggestion
                raise ContentHostError('Test connection failed via insights.')
```

**Review by synkd — APPROVED** [review]

**Comment by jameerpathan111** [comment]
See PRT job 3952 for test results.

File: robottelo/hosts.py (hunk 1 above)
Comment (jameerpathan111) [reply robottelo/hosts.py]: updated

**Review by rmynar — APPROVED** [review]

**Review by Gauravtalreja1 — APPROVED** [review]

**Review by chris1984 — APPROVED** [review]

**Pull request merged**